hub/core/chunk_engine/read.py

```diff
@@ -23,7 +21,7 @@ def read_dataset_meta(storage: StorageProvider):
     return pickle.loads(storage[constants.META_FILENAME])
 
 
-def key_exists(key: str, storage: StorageProvider):
+def tensor_exists(key: str, storage: StorageProvider):
     meta_key = get_meta_key(key)
     index_map_key = get_index_map_key(key)
     return meta_key in storage or index_map_key in storage
```

Comment:
Should this be an "and"? What happens if we have one but not the other?

Reply:
good catch